src/k8s/pkg/k8sd/api/cluster_tokens.go

```diff
@@ -40,6 +40,26 @@ func postClusterJoinTokens(m *microcluster.MicroCluster, s *state.State, r *http
 	return response.SyncResponse(true, &apiv1.TokensResponse{EncodedToken: token})
 }
 
+func getOrCreateJoinTocken(m *microcluster.MicroCluster, tokenName string) (string, error) {
+	// grab token if it exists and return it
+	records, err := m.ListJoinTokens()
+	if err != nil {
+		return "", err
```

Comment:
if we can't list the existing tokens, might as well try to create a new one instead of returning the error.

Reply:
for example, you can log the error here and do `m.NewJoinToken` instead. If I am creating a token, I don't expect to get a "failed to list tokens" error instead.